summingbird-storm/src/main/scala/com/twitter/summingbird/storm/collector/AggregatorOutputCollector.scala

```diff
@@ -46,27 +53,33 @@ class AggregatorOutputCollector[K, V: Semigroup](
     flushExecTimeCounter.incrBy(Time.now.inMillis - startTime.inMillis)
   }
 
-  private def convertToSummerInputFormat(flushedCache: CMap[_, _]): CMap[Int, CMap[_, _]] =
-    flushedCache.groupBy { case (k, _) => summerShards.summerIdFor(k) }
+  private def convertToSummerInputFormat(flushedCache: Map[K, (OutputMessageId, V)]): TraversableOnce[OutputTuple] =
+    flushedCache.groupBy {
+      case (k, _) => summerShards.summerIdFor(k)
+    }.map {
```

Comment:
still think this should be `.iterator.map` to avoid an extra in-memory generation.